tstest/natlab/natlab.go

```diff
@@ -321,59 +332,86 @@ func (m *Machine) hasv6() bool {
 	return false
 }
 
-func (m *Machine) registerConn(c *conn) error {
+func (m *Machine) registerConn4(c *conn) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	if c.ipp.IP.Is6() && c.ipp.IP != v6unspec {
+		return fmt.Errorf("registerConn4 got IPv6 %s", c.ipp)
+	}
+	if _, ok := m.conns4[c.ipp]; ok {
```

Comment:
at least from here down can be shared with registerConn6 with a new helper `registerConn(m *map[net.IPPort]*conn, ipp net.IPPort, c *conn)`